### ISSUE TITLE
latest version hated by windows defender

### ISSUE THREAD

**Issue by mrwiggl** [cadon/ARKStatsExtractor#1102]
Usually the system just says it doesn't think it's safe but lets me continue to use it.  Now it's proactively stopping me from running it completely and deletes the exes in the folder. 
Windows Defender is seeing as a threat :  Trojan:Win32/Emali.A!cl

If I reinstall 0.39.5.1 I can use it again, if I let it update windows shuts it down.

**Comment by VolatilePulse** [comment]
I just realized that you're a couple of versions behind so it is difficult to assess which update is specifically triggering Defender. The latest release version is v0.40.5. It's likely that Defender has a new virus definition added that heuristically matches ASB cause a false positive. When the latest version was released, Defender didn't flag it for me and manually downloading the latest version didn't trigger a Defender warning so it leads me to believe that a recent update has caused this issue with Defender.

As always, we believe that ASB is safe and free of viruses, Trojans, and other malicious software, however, you can never be too safe these days. As long as you have downloaded ASB from GitHub, we feel it's safe to use. When I get home this evening, I will run it through some additional virus scanners to see if something managed to make it's way into this release, but I doubt it will result in anything more than false positives as we have encountered previously.

Some additional background, heuristic definitions are loosely defined and are designed to catch more false positives than not as a new malicious threat has been discovered and not 100% understood immediately. While security researchers investigate viruses and such more thoroughly, they will re-evaluate the virus definition and create a more accurate definition removing a significant number of false positives. Previous versions of ASB that were flagged by a large portion of virus scanners are no longer flagged as the heuristic definition has been changed in those scanners. Unfortunately we deal with this constantly, mostly due to ASB being unsigned software.

**Comment by VolatilePulse** [comment]
The Trojan that Defender is flagging ASB as being, Win32/Emali.a!cl, appears to be related to screen capture software, which ASB includes for OCR support. From all of the versions since v0.39.5.1, it looks like the only modification was the change in v0.40.5 to some OCR features. I wouldn't have expected these changes to suddenly cause ASB to be flagged by Defender, but again, I will look into this further. I'll leave the issue open until I have had a chance to test it.

**Comment by mrwiggl** [comment]
I'm not worried about it being a virus, I am assuming Windows is just looking at features built in as sketchy because the program does have features that are quite different from your usual program.  I did'nt think to just download the latest copy, I'll try that.

**Comment by mrwiggl** [comment]
Just tried downloading the latest version and the download was interupted over thinking it found a virus.

**Comment by VolatilePulse** [comment]
So I was curious and just downloaded the latest portable version and Defender didn't stop me. It also didn't flag the one I recently downloaded. 
![image](https://user-images.githubusercontent.com/6995232/89366790-8f478b80-d6a5-11ea-9f5d-094ae951f910.png)

However, I was curious and checked for Windows updates and sure enough there was an update for Defender so I have it running a scan on the same directory to try and reproduce the issue. And it did not flag either of the existing files.
![image](https://user-images.githubusercontent.com/6995232/89366875-b69e5880-d6a5-11ea-8ce1-09a88d370ae5.png)

VirusTotal's scan of the portable version: https://www.virustotal.com/gui/file/34ec288cc222df505cd449c2ed4cb4b953a066351823b3683736adf455eb82cd/detection
VirusTotal's scan of the installer:
https://www.virustotal.com/gui/file/5b8557a0970d167fb7ab805af48f90cd614c7ef4bb9d58d726085d806a9a0c8e/detection

I'm not quite sure why you're experiencing issues with Defender at the moment as it's working fine for me and I'm not even getting a significant number of hits from VirusTotal (1 hit from each is typically just false positives due to heuristic definitions). Can you provide a screenshot of the Defender protection history for the ASB file you've downloaded? Also, which version have you been having issues with, the portable version or installer? Does the issue exist for the other? What virus definition are you using for Defender? Here's mine:
```
Security intelligence version: 1.321.624.0
Version created on: 8/4/2020 11:10 AM
Last update: 8/4/2020 10:53 PM
```

**Comment by VolatilePulse** [comment]
I'm going to close this issue as I can't reproduce this issue, but feel free to continue to respond and see if we can't figure out why you're experiencing the issue while I can't.

**Comment by JM4nG** [comment]
> Just tried downloading the latest version and the download was interupted over thinking it found a virus.

![smartbreederwarning](https://user-images.githubusercontent.com/69223004/89368654-68c63680-d71f-11ea-96a7-bc4ddb054a32.png)

![image](https://user-images.githubusercontent.com/69223004/89368680-77ace900-d71f-11ea-83b0-51a0bd059891.png)

**Comment by mrwiggl** [comment]
![image](https://user-images.githubusercontent.com/69221938/89368764-28b87280-d691-11ea-9f84-b4d009560580.png)


version 0.40.2.1 installs ok, but soon as i update it - 
![image](https://user-images.githubusercontent.com/69221938/89368912-8baa0980-d691-11ea-8378-22bf05553b35.png)

**Comment by JM4nG** [comment]
I had no issues until I updated some libraries like the better dinos one.

**Comment by VolatilePulse** [comment]
They wouldn't cause this issue. Coincidentally, as soon as both of you responded, Defender finally flagged the updater so I'm going to assume it was a virus definition change they released about 12 hours ago that caused this problem. As mentioned previously, we do not believe there is any malicious code within ASB, however like any software these days, use at your own risk. 

Testing the updater with [VirusTotal](https://www.virustotal.com/gui/file/10a243fdb1c8fd0e5e9c4147791b27e9a512121e2f4603cd8f1d9241b3bfab13/detection) actually shows that Defender flags it. Again, I believe this is a false positive and only appears to be happening after this latest virus definition update. The updater for ASB hasn't been changed recently so it shouldn't suddenly contain a virus.

**Comment by mrwiggl** [comment]
Yes, both the application and the updater app both pop for defender for the same virus definition.  Any changes to classes used or new ones added in?

**Comment by VolatilePulse** [comment]
A custom scan doesn't flag the files for some reason, however, Defender flags it after some time, even when it's not being used. Like I said, the latest version is the most likely culprit due to mild OCR changes, but nothing that should trigger any AV software. It's also curious that Defender didn't flag it for me for several days, but as soon as I update the virus definition (released this morning), it begins flagging the updater. 

I've managed to narrow down the issue to an update to the updater, dating back to version [v0.40.0](https://github.com/cadon/ARKStatsExtractor/releases/tag/v0.40.0) of ASB. The updater, which is currently on version 1.3.0.0, was modified to support TLS 1.2. This version was released on June 1st, 2020 and hasn't changed since. While playing with the updater and VirusTotal, I wasn't able to get the ASB v0.40.0 version of the updater to get flagged by defender, but nothing has changed in the code for the updater since then. There may be some correlation to when the updater was compiled, the size of the compiler, etc. Many of the hashes are identical, but since the updater is compiled when the rest of ASB is compiled, it will always have unique hashes for MD5, SHA-1 & SHA-256, and others. I'm confident that this is still the updater that shipped with v0.40.0 and hasn't been modified.

I'm out of ideas as to why this has been flagged randomly though.

**Comment by JM4nG** [comment]
here is a megatherium I cant seem to get right in the calculator
